staging/src/k8s.io/client-go/dynamic/fake/simple_test.go

```diff
@@ -181,21 +181,15 @@ func TestPatch(t *testing.T) {
 			patchBytes: []byte(`[{"op": "add", "path": "/spec/newvalue", "value": "dummy"}]`),
 			wantErrMsg: "invalid JSON document",
 		}, {
-			name:                  "strategic merge is no-op with no changes",
-			object:                newUnstructuredWithSpec(map[string]interface{}{"foo": "bar"}),
-			patchType:             types.StrategicMergePatchType,
-			patchBytes:            []byte(`{}`),
-			expectedPatchedObject: newUnstructuredWithSpec(map[string]interface{}{"foo": "bar"}),
-		}, {
-			name:                  "strategic merge change",
-			object:                newUnstructured(testAPIVersion, testKind, testNamespace, testName),
-			patchType:             types.StrategicMergePatchType,
-			patchBytes:            []byte(`{"kind": "newkind"}`),
-			expectedPatchedObject: newUnstructured(testAPIVersion, "newkind", testNamespace, testName),
+			name:       "strategic merge fails",
+			object:     newUnstructured(testAPIVersion, testKind, testNamespace, testName),
+			patchType:  types.MergePatchType,
+			patchBytes: []byte(`{}`),
+			wantErrMsg: "PatchType is not supported",
```

Comment:
Can you add a TODO and open a follow-up issue to add test for strategic merge patch?

Reply:
https://github.com/kubernetes/client-go/issues/488